pkg/es/config/config.go

```diff
@@ -142,6 +142,9 @@ type Configuration struct {
 	// latest adaptive sampling probabilities.
 	AdaptiveSamplingLookback time.Duration `mapstructure:"adaptive_sampling_lookback"`
 	Tags                     TagsAsFields  `mapstructure:"tags_as_fields"`
+	// IsOpenSearch stores whether the backend is of opensearch type or not.
+	// If kept empty, jaeger will automatically identify the distinction.
+	IsOpenSearch bool `mapstructure:"is_open_search"`
```

Comment:
we never needed this, why now? We can automatically determine the backend based on the info it returns from the initial handshake.